### ISSUE TITLE
Freeze package versions

### ISSUE THREAD

**Issue by AmirSavand** [Southclaws/sampctl#269]
When having multiple packages installed without specifying tag versions which means every time sampctl installs them, it pulls the latest version.

That's bad for most projects and the current solution is to set all packages version tag manually which is fine but if there is a freeze command that can do this, it'll be great.

Here's `pip freeze` information: https://pip.pypa.io/en/stable/reference/pip_freeze/

Note that `pip freeze` doesn't update any files it just prints the packages with current installed version out (to save them: `pip freeze >> requirements.txt`) but in sampctl it should update packages too.

**Comment by AmirSavand** [comment]
I don't have much experience with sampctl so if I'm missing something here then let me know.

**Comment by Southclaws** [comment]
Ah yes, I've been meaning to add this feature for a while now. I wanted to make this the default whenever a user installed a package without any constraint, sampctl would simply choose the latest *tagged* version (if any) and if there aren't any tags, it would simply apply a commit hash constraint.

Eventually, I'd like to remove un-versioned dependencies entirely so it's impossible to specify a dependency without it being versioned in some way.

**Comment by AmirSavand** [comment]
Yes, that's what NPM does too, it sets the tag of latest version if you don't specify it when installing.
